common/lib/client/realtimechannel.js

```diff
@@ -423,22 +438,49 @@ var RealtimeChannel = (function() {
 
 		case actions.MESSAGE:
 			var messages = message.messages,
+				firstMessage = messages[0],
+				lastMessage = messages[messages.length - 1],
 				id = message.id,
 				connectionId = message.connectionId,
 				timestamp = message.timestamp;
 
+			if(this.decodeFailureRecoveryInProgress) {
+				if(message.channelSerial === this.decodeFailureRecoveryForChannelSerial) {
+					this.decodeFailureRecoveryInProgress = false;
+					this.decodeFailureRecoveryForChannelSerial = null;
+				} else {
+					Logger.logAction(Logger.LOG_MAJOR, 'RealtimeChannel.onMessage()', 'Discarding message received during delta failure recovery process.');
+					break;
+				}
```

Comment:
I don't think this is right -- the channel can't be stuck in `decodeFailureRecoveryInProgress=true` unless/until you get a message with the requested channelSerial and nothing else. E.g. what if the attach fails to recover back to that channelSerial and just gives you the current position? Or any other reason that you might not get a message with the expect channel serial (e.g. the library becomes disconnected for > 2 minutes during this process). With what you have here the channel would then be permanently refusing to process messages, even if it would be able to fine.

I think you have to keep trying to process all messages that arrive on the basis of what you have; the only effect that a recovery in progress should make is that you don't send a second attach while there's one in progress, and that state should cease to hold the moment you get any attached or detached (or error)

Reply:
Fixed in https://github.com/ably/ably-js/pull/615/commits/b0cc789cbe06b3441aa5c2a0c6af8c4c514abb45